case-lib/lib.sh

```diff
@@ -952,7 +952,7 @@ is_ipc4()
 logger_disabled()
 {
     # Disable logging when available...
-    if [ ${OPT_VAL['s']} -eq 0 ]; then
+    if [[ ${OPT_VAL['s']} -eq 0 ]]; then
```

Comment:
That difference between `[ ]` and `[[  ]]` is too subtle IMHO, I think most people won't know it. You could just do this instead:

```suggestion
    if [ "${OPT_VAL['s']}" = 0 ]; then
```

Reply:
This should be correct but please test it:

```suggestion
    if [ -n "${OPT_VAL['s']}" ] && [ "${OPT_VAL['s']}" -eq 0 ]; then
```

Reply:
your both suggestions are wrong, both return
```
-bash: 's': syntax error: operand expected (error token is "'s'")
```
[[ ... ]] - it fixed current error in this script. But you are right an undefined and a default value should enable sof_logger, it's easy to fixed  some tests have set default value and for other we can check if ['s'] is set if not we can set to 1.